### PR TITLE
Update sticky settings docs to flesh out external content

### PR DESCRIPTION
**Pull request by blackbaud-johnly**
Addresses https://github.com/blackbaud/skyux2-docs/issues/583. (A PR at https://blackbaud.visualstudio.com/Products/_git/skyux-spa-ui-config-service-docs/pullrequest/66211?_a=overview has corresponding updates to the internal UI Config site to remove the content in that site that was ported over to this site.)

### CODE REVIEW

File: src/app/components/flyout/index.html [hunk 1]
```diff
@@ -102,7 +102,7 @@
       propertyName="settingsKey"
       isOptional="true"
     >
-      Specifies a unique key for the UI Config Service to retrieve stored settings from a database. The UI Config Service saves configuration settings for users to preserve the width of the flyout. This property accepts <sky-code>string</sky-code> values. This property is for internal Blackbaud use only. For Blackbaud developers, the UI Config Service website describes how to <a href="https://docs.blackbaud.com/ui-config-service-docs/learn/integrations/skyux-components/flyouts">apply sticky settings to flyouts</a>.
+      Specifies a unique key for the UI Config Service to retrieve stored settings from a database. The UI Config Service saves configuration settings for users to preserve the width of the flyout. This property accepts <sky-code>string</sky-code> values. This property is for internal Blackbaud use only. For more information about the UI Config Service, see <a stacheRouterLink="/learn/get-started/advanced/sticky-settings">the sticky settings documentation</a>.
```
Comment (Blackbaud-BobbyEarl) [inline src/app/components/flyout/index.html]: We should remove the "This property is for internal Blackbaud use only."

File: src/app/components/grid/index.html [hunk 2]
```diff
@@ -82,7 +82,7 @@
...
       isOptional="true"
     >
-    Specifies a unique key for the UI Config Service to retrieve stored settings from a database. The UI Config Service saves configuration settings for users and returns <sky-code>selectedColumnIds</sky-code> to preserve their preferred column order and the sorting of columns. This property accepts <sky-code>string</sky-code> values. This property is for internal Blackbaud use only. For Blackbaud developers, the UI Config Service website describes how to <a href="https://docs.blackbaud.com/ui-config-service-docs/learn/integrations/skyux-components/grids">apply sticky settings to grids</a>.
+    Specifies a unique key for the UI Config Service to retrieve stored settings from a database. The UI Config Service saves configuration settings for users and returns <sky-code>selectedColumnIds</sky-code> to preserve their preferred column order and the sorting of columns. This property accepts <sky-code>string</sky-code> values. This property is for internal Blackbaud use only. For more information about the UI Config Service, see <a stacheRouterLink="/learn/get-started/advanced/sticky-settings">the sticky settings documentation</a>
```
Comment (Blackbaud-BobbyEarl) [inline src/app/components/grid/index.html]: We should remove the "This property is for internal Blackbaud use only."

File: src/app/components/tile/index.html [hunk 3]
```diff
@@ -71,7 +71,7 @@
       propertyName="settingsKey"
       isOptional="true"
     >
-    Specifies a unique key for the UI Config Service to retrieve stored settings from a database. The UI Config Service saves configuration settings for users to preserve the layout and collapsed state of tile dashboards. This property accepts <sky-code>string</sky-code> values. This property is for internal Blackbaud use only. For Blackbaud developers, the UI Config Service website describes how to <a href="https://docs.blackbaud.com/ui-config-service-docs/learn/integrations/skyux-components/tiles">apply sticky settings to tile dashboards</a>.
+    Specifies a unique key for the UI Config Service to retrieve stored settings from a database. The UI Config Service saves configuration settings for users to preserve the layout and collapsed state of tile dashboards. This property accepts <sky-code>string</sky-code> values. This property is for internal Blackbaud use only. For more information about the UI Config Service, see <a stacheRouterLink="/learn/get-started/advanced/sticky-settings">the sticky settings documentation</a>
```
Comment (Blackbaud-BobbyEarl) [inline src/app/components/tile/index.html]: We should remove the "This property is for internal Blackbaud use only."

File: src/app/learn/get-started/advanced/sticky-settings/integrations/custom-components/index.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
+<stache
+  pageTitle="Custom Components"
+  navTitle="Custom components"
+  navOrder="20">
+
+  <p>
+    To save user configuration settings in custom components, use the <sky-code>SkyUIConfigService</sky-code> as illustrated in the following code sample. The <sky-code>key</sky-code> that you provide to the <sky-code>SkyAppUIConfigService</sky-code> must be unique within your SPA.
+  </p>
+
```
Comment (Blackbaud-BobbyEarl) [inline src/app/learn/get-started/advanced/sticky-settings/integrations/custom-components/index.html]: I think this needs the same warning as the SKY UX components pages... "For Blackbaud developers, the settingsKey must be added to a server-side whitelist. Blackbaud developers can learn how to add keys to that whitelist on the UI Config Service website."

File: src/app/learn/get-started/advanced/sticky-settings/overview/index.html [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  LOCAL_STORAGE,
+  StorageService
+} from 'ngx-webstorage-service';
+
+import {
+  SkyAppConfig
+} from '@skyux/config';
+
+@Injectable()
+export class SkyAppUIConfigService {
```
Comment (Blackbaud-BobbyEarl) [inline src/app/learn/get-started/advanced/sticky-settings/overview/index.html]: Let's rename this class to 'AppUIConfigService`.

File: src/app/learn/get-started/advanced/sticky-settings/overview/index.html [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  <sky-alert>
+    For Blackbaud developers, the sticky settings functionality is available first-class through the UI Config Service that is for internal Blackbaud use only. Blackbaud developers can learn how to implement sticky settings on <a href="https://docs.blackbaud.com/ui-config-service-docs/">the UI Config Service website</a>.
+  </sky-alert>
+  <p>
+    To implement sticky settings, you must provide a class for <sky-code>SkyUIConfigService</sky-code> in your SPA's <sky-code>app-extras.module.ts</sky-code> file.
+  </p>
+  <sky-code-block languageType="typescript">
+    import {
+      AppUIConfigService
```
Comment (Blackbaud-BobbyEarl) [inline src/app/learn/get-started/advanced/sticky-settings/overview/index.html]: Let's rename this class to 'AppUIConfigService`.

File: src/app/learn/get-started/advanced/sticky-settings/index.html [hunk 7]
```diff
@@ -5,98 +5,13 @@
 
   <stache-page-summary>
     <p>
-      Sticky settings enable you to save user settings across sessions in grids and tile dashboards. Sticky settings preserve column order and the sorting of columns in grids, and they preserve the layout and collapsed state of tiles in tile dashboards.
+      Sticky settings enable you to save user settings across sessions in flyouts, grids, and tile dashboards.
```
Comment (Blackbaud-BobbyEarl) [inline src/app/learn/get-started/advanced/sticky-settings/index.html]: Maybe add "and custom components/services," or something like that?

File: src/app/learn/get-started/advanced/sticky-settings/integrations/skyux-components/flyouts/index.html [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    </p>
+    <stache-include fileName="sticky-settings-whitelist.html"></stache-include>
+
+    <sky-code-block languageType="typescript">
+       public openFlyoutWithCutsomWidth(): void {
+        const flyoutConfig: SkyFlyoutConfig = {
+          defaultWidth: 350,
+          maxWidth: 500,
+          minWidth: 200,
+          settingsKey: my-flyout
```
Comment (Blackbaud-BobbyEarl) [inline src/app/learn/get-started/advanced/sticky-settings/integrations/skyux-components/flyouts/index.html]: `my-flyout` needs quotes.

File: src/app/components/flyout/index.html (hunk 1 above)
Comment (blackbaud-johnly) [reply src/app/components/flyout/index.html]: Good catch. Fixed this locally and will push the change shortly.

File: src/app/components/grid/index.html (hunk 2 above)
Comment (blackbaud-johnly) [reply src/app/components/grid/index.html]: Good catch. Fixed this locally and will push the change shortly.

File: src/app/components/tile/index.html (hunk 3 above)
Comment (blackbaud-johnly) [reply src/app/components/tile/index.html]: Good catch. Fixed this locally and will push the change shortly.

File: src/app/learn/get-started/advanced/sticky-settings/integrations/custom-components/index.html (hunk 4 above)
Comment (blackbaud-johnly) [reply src/app/learn/get-started/advanced/sticky-settings/integrations/custom-components/index.html]: Good catch. Added this locally and will push the change shortly.

File: src/app/learn/get-started/advanced/sticky-settings/index.html (hunk 7 above)
Comment (blackbaud-johnly) [reply src/app/learn/get-started/advanced/sticky-settings/index.html]: I added ", as well as in custom components and services" to the end of the sentence and will push the changes up shortly.

File: src/app/learn/get-started/advanced/sticky-settings/integrations/skyux-components/flyouts/index.html (hunk 8 above)
Comment (blackbaud-johnly) [reply src/app/learn/get-started/advanced/sticky-settings/integrations/skyux-components/flyouts/index.html]: Good catch. Added this locally and will push the change shortly.

File: src/app/learn/get-started/advanced/sticky-settings/overview/index.html (hunk 5 above)
Comment (blackbaud-johnly) [reply src/app/learn/get-started/advanced/sticky-settings/overview/index.html]: Good catch. I removed "Sky."

File: src/app/learn/get-started/advanced/sticky-settings/overview/index.html (hunk 6 above)
Comment (blackbaud-johnly) [reply src/app/learn/get-started/advanced/sticky-settings/overview/index.html]: I removed "Sky" in line 31 as discussed offline.

File: src/app/learn/get-started/advanced/sticky-settings/integrations/custom-components/index.html [hunk 9]
```diff
@@ -0,0 +1,53 @@
+<stache
+  pageTitle="Custom Components"
+  navTitle="Custom components"
+  navOrder="20">
+
+  <p>
+    To save user configuration settings in custom components, use the <sky-code>SkyUIConfigService</sky-code> as illustrated in the following code sample. The <sky-code>key</sky-code> that you provide to the <sky-code>SkyAppUIConfigService</sky-code> must be unique within your SPA.
```
Comment (Blackbaud-BobbyEarl) [inline src/app/learn/get-started/advanced/sticky-settings/integrations/custom-components/index.html]: `SkyAppUIConfigService` should be `SkyUIConfigService`.

File: src/app/learn/get-started/advanced/sticky-settings/integrations/skyux-components/flyouts/index.html [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    <p>
+      To save user configuration settings in flyouts, add the <sky-code>settingsKey</sky-code> property to the <sky-code>config</sky-code> parameter of the <sky-code>open(component, config?)</sky-code> property that the flyout service uses to open the flyout. The <sky-code>settingsKey</sky-code> must be unique to the component.
+    </p>
+    <p>
+      The UI Config Service preserves the width of flyouts.
+    </p>
+    <stache-include fileName="sticky-settings-whitelist.html"></stache-include>
+
+    <sky-code-block languageType="typescript">
+       public openFlyoutWithCutsomWidth(): void {
```
Comment (Blackbaud-BobbyEarl) [inline src/app/learn/get-started/advanced/sticky-settings/integrations/skyux-components/flyouts/index.html]: Formatting for this demo is slightly off... should be:

```
public openFlyoutWithCustomWidth(): void {
  const flyoutConfig: SkyFlyoutConfig = {
    defaultWidth: 350,
    maxWidth: 500,
    minWidth: 200,
    settingsKey: 'my-flyout'
  };

  this.flyout = this.flyoutService.open(SkyFlyoutDemoInternalSimpleComponent, flyoutConfig);
}

**Pull request merged**